eclass/kernel-build.eclass

```diff
@@ -369,6 +388,94 @@ kernel-build_src_install() {
 		secureboot_sign_efi_file "${ED}${kernel_dir}/${image_path}"
 	fi
 
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]] && use generic-uki; then
+		local image="${ED}${kernel_dir}/${image_path}"
+
+		# NB: if you pass a path that does not exist or is not a regular
+		# file/directory, dracut will silently ignore it and use the default
+		# https://github.com/dracutdevs/dracut/issues/1136
+		> "${T}"/empty-file || die
+		mkdir -p "${T}"/empty-directory || die
+
+		local dracut_modules=(
+			base bash btrfs cifs crypt crypt-gpg crypt-loop dbus dbus-daemon dm
+			dmraid drm dracut-systemd fido2 i18n iscsi fs-lib kernel-modules
+			kernel-network-modules kernel-modules-extra lunmask lvm nbd mdraid
+			modsign network network-manager nfs nvdimm nvmf pcsc pkcs11 plymouth
+			qemu qemu-net resume rngd rootfs-block shutdown systemd
+			systemd-ac-power systemd-ask-password systemd-initrd
+			systemd-integritysetup systemd-pcrphase systemd-sysusers
+			systemd-udevd systemd-veritysetup terminfo tpm2-tss udev-rules
+			uefi-lib usrmount virtiofs
+		)
+
+		local dracut_args=(
+			--conf "${T}/empty-file"
+			--confdir "${T}/empty-directory"
+			--kernel-image "${image}"
+			--kmoddir "${ED}/lib/modules/${dir_ver}"
+			--kver "${dir_ver}"
+			--verbose
+			--no-hostonly
+			--no-hostonly-cmdline
+			--no-hostonly-i18n
+			--no-machineid
+			--nostrip
+			--no-uefi
+			--early-microcode
+			--reproducible
+			--ro-mnt
+			--modules "${dracut_modules[*]}"
+		)
+
+		# Tries to update ld cache
+		addpredict /etc/ld.so.cache~
```

Comment:
I'm confused why you only need `~`.

Reply:
> I'm confused why you only need `~`.

Same, I also don't really understand this. I noticed that the exact same error (access denied to `/etc/ld.so.cache~`)  happens when we build the testing dracut image in `src_test`, it just never was fatal there because it appears that `src_test` already has `addpredict /etc/ld.so.cache~` set somewhere. I couldn't find where this `src_test` `addpredict /etc/ld.so.cache~` comes from though, perhaps somewhere in the portage code.